`TestTSHConfigConnectWithOpenSSHClient` flakiness

## Failure
saw this fail off of branch/v10

### CI Logs
- https://console.cloud.google.com/cloud-build/builds/8cd306c3-13b3-454f-bbb3-f3ccf70f797b?project=ci-account
- https://console.cloud.google.com/cloud-build/builds;region=us-west1/d926b469-8a79-41a0-891f-bcb54f9b8ae1;step=0?project=ci-account

### Relevant snippet
```
proxy_test.go:590: failed to find AuthAttemptFailureCode event
    --- FAIL: TestTSHConfigConnectWithOpenSSHClient/node_recording_mode_with_TLS_routing_enabled (6.28s)
```

--- a/tool/tsh/common/proxy_test.go
+++ b/tool/tsh/common/proxy_test.go
@@ -1100,14 +1100,15 @@ func mustSearchEvents(t *testing.T, auth *auth.Server) []apievents.AuditEvent {
 }
 
 func mustFindFailedNodeLoginAttempt(t *testing.T, s *suite, nodeLogin string) {
-	av := mustSearchEvents(t, s.root.GetAuthServer())
-	for _, e := range av {
-		if e.GetCode() == events.AuthAttemptFailureCode {
-			require.Equal(t, e.(*apievents.AuthAttempt).Login, nodeLogin)
-			return
+	require.Eventually(t, func() bool {
+		es := mustSearchEvents(t, s.root.GetAuthServer())
+		for _, e := range es {
+			if e.GetCode() == events.AuthAttemptFailureCode && e.(*apievents.AuthAttempt).Login == nodeLogin {
+				return true
+			}
 		}
-	}
-	t.Errorf("failed to find AuthAttemptFailureCode event (0/%d events matched)", len(av))
+		return false
+	}, 5*time.Second, 500*time.Millisecond, "failed to find AuthAttemptFailureCode event")
 }
 
 func TestFormatCommand(t *testing.T) {

--- a/tool/tsh/common/proxy_test.go
+++ b/tool/tsh/common/proxy_test.go
@@ -1098,11 +1098,12 @@ func mustFindFailedNodeLoginAttempt(t *testing.T, s *suite, nodeLogin string) {
 		assert.NoError(t, err)
 
 		for _, e := range es {
-			if e.GetCode() == events.AuthAttemptFailureCode && e.(*apievents.AuthAttempt).Login == nodeLogin {
+			if e.GetCode() == events.AuthAttemptFailureCode {
+				assert.Equal(t, e.(*apievents.AuthAttempt).Login, nodeLogin)
 				return
 			}
 		}
-		t.Errorf("Failed to find an AuthAttemptFailureCode event")
+		t.Errorf("failed to find AuthAttemptFailureCode event (0/%d events matched)", len(es))
 	}, 5*time.Second, 500*time.Millisecond)
 }
 

--- a/tool/tsh/common/proxy_test.go
+++ b/tool/tsh/common/proxy_test.go
@@ -1086,29 +1086,24 @@ func mustFailToRunOpenSSHCommand(t *testing.T, configFile string, sshConnString 
 	require.Error(t, err)
 }
 
-func mustSearchEvents(t *testing.T, auth *auth.Server) []apievents.AuditEvent {
-	now := time.Now()
-	ctx := context.Background()
-	events, _, err := auth.SearchEvents(ctx, events.SearchEventsRequest{
-		From:  now.Add(-time.Hour),
-		To:    now.Add(time.Hour),
-		Order: types.EventOrderDescending,
-	})
-
-	require.NoError(t, err)
-	return events
-}
-
 func mustFindFailedNodeLoginAttempt(t *testing.T, s *suite, nodeLogin string) {
-	require.Eventually(t, func() bool {
-		es := mustSearchEvents(t, s.root.GetAuthServer())
+	require.EventuallyWithT(t, func(t *assert.CollectT) {
+		now := time.Now()
+		ctx := context.Background()
+		es, _, err := s.root.GetAuthServer().SearchEvents(ctx, events.SearchEventsRequest{
+			From:  now.Add(-time.Hour),
+			To:    now.Add(time.Hour),
+			Order: types.EventOrderDescending,
+		})
+		assert.NoError(t, err)
+
 		for _, e := range es {
 			if e.GetCode() == events.AuthAttemptFailureCode && e.(*apievents.AuthAttempt).Login == nodeLogin {
-				return true
+				return
 			}
 		}
-		return false
-	}, 5*time.Second, 500*time.Millisecond, "failed to find AuthAttemptFailureCode event")
+		t.Errorf("Failed to find an AuthAttemptFailureCode event")
+	}, 5*time.Second, 500*time.Millisecond)
 }
 
 func TestFormatCommand(t *testing.T) {

